### PR TITLE
Always display ranged/melee toggle, notify if situational

### PR DESCRIPTION
**Pull request by melchj**
With these changes, the Melee/Ranged damage toggle will be present for all damaging spells, regardless of their min/max range. A notification is present for the damage type that is atypical.

See issue #349.

Previously, spells with min range >1 would only show ranged damage, and spells with max range <2 would only show melee damage. However, with the use of elio portals, area of effect, and other situational possibilities, any spell could potentially deal ranged or melee damage.

Example of tooltip on desktop (warning is only visible when the melee/ranged toggle is selecting the atypical damage type):
image
image

On mobile:
image

### CODE REVIEW

File: client/common/wrappers.tsx [hunk 1]
```diff
@@ -443,11 +458,13 @@ export const DamageTypeToggle: React.FC<{
     <div
       css={{
         display: 'flex',
+        alignItems: 'center',
```
Comment (lightyearsaway) [inline client/common/wrappers.tsx:461]: Nice, appreciate the attention to detail!

File: client/common/wrappers.tsx [hunk 2]
```diff
@@ -417,10 +418,24 @@ export const CustomSetHead: React.FC<{ customSet?: CustomSet | null }> = ({
...
   readonly setShowRanged: React.Dispatch<React.SetStateAction<boolean>>;
-}> = ({ showRanged, setShowRanged }) => {
+  readonly rangedOnly: boolean;
+  readonly meleeOnly: boolean;
+}> = ({ showRanged, setShowRanged, rangedOnly, meleeOnly }) => {
   const { t } = useTranslation('weapon_spell_effect');
   const theme = useTheme();
 
+  const notPossibleWarning = (
+    <span css={{ marginLeft: 8}}>
```
Comment (lightyearsaway) [inline client/common/wrappers.tsx:428]: Could you setup prettier if you don't have it or check your configuration? I have it set to format on save, it's amazing

File: client/common/wrappers.tsx [hunk 3]
```diff
@@ -417,10 +418,24 @@ export const CustomSetHead: React.FC<{ customSet?: CustomSet | null }> = ({
...
+        <Tooltip
+          css={{ textAlign: 'center' }}
+          title={meleeOnly ? t('RANGED_NOT_POSSIBLE') : t('MELEE_NOT_POSSIBLE')}
+          arrowPointAtCenter={true}
+        >
+          <FontAwesomeIcon icon={faExclamationCircle} />
```
Comment (lightyearsaway) [inline client/common/wrappers.tsx:434]: Do you think we should consider using the same yellow triangle as we do for warnings for when item conditions are not met?

**Review by lightyearsaway — COMMENTED** [review]
Looks good! Just a couple nits

File: client/common/wrappers.tsx (hunk 2 above)
Comment (melchj) [reply client/common/wrappers.tsx:428]: Good catch, looks like I didn't have it set up. Formatting on save now, good idea

File: client/common/wrappers.tsx (hunk 3 above)
Comment (melchj) [reply client/common/wrappers.tsx:434]: Yes, I like how this fits with the rest of the ui. Here is what it looks like with this change:

desktop:
image

mobile:
image

**Comment by melchj** [comment]
Commit 7c08ee7 added to resolve above comments.

**Pull request merged**